bin/rule.ml

```diff
@@ -29,9 +29,18 @@ let prepend_root r b = match r with
   | None   -> b
   | Some r -> Filename.concat r b
 
+let pp_dep fmt = function
+  | `Named (name, path) -> Fmt.pf fmt "(:%s %s)" name path
```

Comment:
Is there any particular reason to use polymorphic variants here? I think I'd prefer we define types for dependencies and actions here.

Reply:
That's because these values are created a few lines below and directly passed to this function. So it's not worth the overhead of defining the type.
What do you think ?

Reply:
I think that if we extract proper types, move those to the library we can then unit test how they are printed rather than using end to end testing for that.

We can probably do that separately though. Let's keep it like this for now, it's already an improvement!